Dapr crashes (panics) when pub-sub subscriber takes long time to respond

## In what area(s)?

/area runtime

## What version of Dapr?
0.6.0



## Actual Behavior

Dapr crashes (panics) when pub-sub subscriber takes long time to respond.


## Steps to Reproduce the Problem

I am running a typical pub-sub scenario.
Publisher is running in Kubernetes and publishing events to Service Bus.
Subscriber event handler is running locally and receiving an event from Service Bus.

I originally discovered the issue when trying step-by-step debugging, but was later able to reproduce it by adding a sleep (100 sec) to my subscriber handler. After ~60 sec daprd panics with the following stack trace:

```
app_id=batch-processor instance=MININT-MKM8KKN scope=dapr.runtime type=log ver=0.6.0
time="2020-04-17T10:08:54.2417907-07:00" level=info msg="dapr initialized. Status: Running. Init Elapsed 6330.2773ms" app_id=batch-processor instance=MININT-MKM8KKN scope=dapr.runtime type=log ver=0.6.0
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x18 pc=0x1435c14]

goroutine 389 [running]:
github.com/dapr/dapr/pkg/runtime.(*DaprRuntime).publishMessageHTTP(0xc0002b8b40, 0xc000b11a70, 0xc0003b4c01, 0xc000b11a70)
        D:/a/dapr/dapr/pkg/runtime/runtime.go:944 +0x374
github.com/dapr/components-contrib/pubsub/azure/servicebus.(*azureServiceBus).getHandlerFunc.func1(0x36012e0, 0xc0003b4cc0, 0xc0004f2000, 0x36012e0, 0xc0003b4cc0)
        C:/Users/runneradmin/go/pkg/mod/github.com/dapr/components-contrib@v0.0.0-20200331025048-fe9b90075a54/pubsub/azure/servicebus/servicebus.go:213 +0xae
github.com/dapr/components-contrib/pubsub/azure/servicebus.(*azureServiceBus).handleSubscriptionMessages.func1.1(0x3601201, 0xc000249080, 0xc0002e6000, 0xc000615780, 0xc0004f2000, 0xc000280360, 0xd)
        C:/Users/runneradmin/go/pkg/mod/github.com/dapr/components-contrib@v0.0.0-20200331025048-fe9b90075a54/pubsub/azure/servicebus/servicebus.go:247 +0xc8
created by github.com/dapr/components-contrib/pubsub/azure/servicebus.(*azureServiceBus).handleSubscriptionMessages.func1      
        C:/Users/runneradmin/go/pkg/mod/github.com/dapr/components-contrib@v0.0.0-20200331025048-fe9b90075a54/pubsub/azure/servicebus/servicebus.go:236 +0x7b
The terminal process terminated with exit code: 2

Terminal will be reused by tasks, press any key to close it.
```

In local environment daprd isn't restarted automatically so nothing works after this point.

--- a/pkg/runtime/runtime.go
+++ b/pkg/runtime/runtime.go
@@ -943,9 +943,20 @@ func (a *DaprRuntime) publishMessageHTTP(msg *pubsub.NewMessage) error {
 	}
 
 	resp, err := a.appChannel.InvokeMethod(&req)
-	statusCode := http.GetStatusCodeFromMetadata(resp.Metadata)
+	statusCode := 0
+
+	if resp != nil {
+		statusCode = http.GetStatusCodeFromMetadata(resp.Metadata)
+	}
+
 	if err != nil || statusCode != 200 {
-		err = fmt.Errorf("error from app while processing pub/sub event: %s. status code returned: %v", string(resp.Data), statusCode)
+		var err error
+		if resp == nil {
+			err = fmt.Errorf("error from app channel while sending pub/sub event to app: %s", err)
+		} else {
+			err = fmt.Errorf("error returned from app while processing pub/sub event: %s. status code returned: %v", string(resp.Data), statusCode)
+		}
+
 		log.Debug(err)
 		return err
 	}

--- a/pkg/runtime/runtime_test.go
+++ b/pkg/runtime/runtime_test.go
@@ -667,7 +667,7 @@ func TestOnNewPublishedMessage(t *testing.T) {
 			Data:     []byte(clientError.Error()),
 		}
 
-		expectedClientError := fmt.Errorf("error from app while processing pub/sub event: Internal Error. status code returned: 500")
+		expectedClientError := fmt.Errorf("error returned from app while processing pub/sub event: Internal Error. status code returned: 500")
 
 		mockAppChannel.On("InvokeMethod", expectedRequest).Return(fakeHTPResponse, clientError)
 

